slither/detectors/statements/unused_imports.py

```diff
@@ -0,0 +1,481 @@
+            for v in vs:
+                to_remove |= all_imports[v]
+            for imp in to_remove:
+                if imp in vs:
+                    vs.remove(imp)
+
+    def _get_needed_but_not_imported_directly(self) -> Dict[str, Set[str]]:
+        """
+        Returns a dict, that for each file, contains files that are needed by it, but not imported directly (via
+        "import" statement). These files can, however, be imported indirectly.
+        For instance, if we have:
+            C.sol
+                import B.sol // unneeded, but something from A.sol is used somewhere in C.sol
+            B.sol:
+                import A.sol
+        then needed_but_not_imported_directly[C.sol] = {A.sol}, even though A.sol is imported indirectly in C.sol using
+        "import B.sol".
+        """
+        needed_but_not_imported_directly = self._new_dict()
+        for k, v in self.needed_imports.items():
+            needed_but_not_imported_directly[k] = v - self.actual_imports[k]
+        return needed_but_not_imported_directly
+
+    def _get_imported_but_unneeded(self) -> Dict[str, Set[str]]:
+        """
+        Returns a dict, that for each file, contains files that are directly imported by it (via "import" statement),
+        but are not needed (even, if their own imports are needed).
+        For instance, if we have:
+            C.sol:
+                import B.sol // unneeded, but something from A.sol is used somewhere in C.sol
+            B.sol:
+                import A.sol
+        then imported_but_unneeded[C.sol] = {B.sol}, even though A.sol is needed and imported indirectly in C.sol using
+        "import B.sol".
+        """
+        imported_but_unneeded = self._new_dict()
+        for k, v in self.actual_imports.items():
+            imported_but_unneeded[k] = v - self.needed_imports[k]
+        return imported_but_unneeded
+
+    def _detect(self):
+        results = []
+
+        self._initialise_absolute_path_to_imp_filename()
+        self._initialise_actual_imports()
+        self._find_top_level_items_uses()
+        self._find_contract_level_items_uses()
+
+        self._detect_cycles(self.actual_imports)
+        if self.imports_cycle_detected:
+            # no support for cycles in import graph at the moment
+            return results
+
+        all_imports = self._get_all_imports()
+        all_imports_in_needed = self._get_all_imports_in_needed(all_imports)
+        self._remove_redundant_imports_from_needed(all_imports)
+
+        needed_but_not_imported_directly = self._get_needed_but_not_imported_directly()
+        imported_but_unneeded = self._get_imported_but_unneeded()
+
+        for k, v in imported_but_unneeded.items():
+            info = []
+            output = ""
+            if k in UnusedImports.analysed_files or len(v) == 0:
+                continue
+            output += (
+                "Unused imports found in "
+                + PurePath(os.path.relpath(k, os.curdir)).as_posix()
```

Comment:
Not 100% sure, but I think we can get rid of the local/CI file path discrepancies by using `PurePath(k).as_posix()` here

Reply:
I have made the change you suggested, however, for some reason, tests weren't run on the code, so I am not sure if it passes the tests that were failing.